src/algorithms/tracking/TrackSeeding.cc

```diff
@@ -188,15 +189,12 @@ std::unique_ptr<edm4eic::TrackParametersCollection> eicrecon::TrackSeeding::make
       auto slopeZ0 = lineFit(rzHitPositions);
       const auto xypos = findPCA(RX0Y0);
 
-      //Determine charge
-      std::vector<std::pair<float,float>> xyrelPos;
+      auto xpos = xypos.first;
+      auto ypos = xypos.second;
 
-      for ( const auto& spptr : seed.sp() )
-      {
-        xyrelPos.emplace_back(spptr->x()-xypos.first, spptr->y()-xypos.second);
-      }
+      //Determine charge
 
-      int charge = determineCharge(xyrelPos);
+      int charge = determineCharge(xyHitPositions, X0, Y0, xpos, ypos);
```

Comment:
Don't pass x and y separate, but combine them in a proper 2D vector before passing, e.g. with `V0{X0, Y0}` and `v{x,y}`:
```suggestion
      int charge = determineCharge(xyHitPositions, V0, v);
```